src/Features/CSharp/Portable/Snippets/CSharpForEachLoopSnippetProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Snippets.SnippetProviders;
+
+namespace Microsoft.CodeAnalysis.CSharp.Snippets
+{
+    [ExportSnippetProvider(nameof(ISnippetProvider), LanguageNames.CSharp), Shared]
+    internal sealed class CSharpForEachLoopSnippetProvider : AbstractForEachLoopSnippetProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpForEachLoopSnippetProvider()
+        {
+        }
+
+        /// <summary>
+        /// Creates the foreach statement syntax.
+        /// Must be done in language specific file since there is no generic way to generate the syntax.
+        /// </summary>
+        protected override async Task<SyntaxNode> CreateForEachLoopStatementSyntaxAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var varIdentifier = SyntaxFactory.IdentifierName("var");
+            var collectionIdentifier = SyntaxFactory.IdentifierName("collection");
+            var itemString = NameGenerator.GenerateUniqueName(
+                "item", n => semanticModel.LookupSymbols(position, name: n).IsEmpty);
```

Comment:
nit: I would expect this to use the overload `GenerateUniqueName(string, IEnumerable<ISymbol>)` instead of doing this check. I'm not sure which is faster, but you're basically duplicating the logic here.